src/vm.rs

```diff
@@ -90,10 +92,27 @@ impl<VCpuType: VirtualCPU> UhyveVm<VCpuType> {
 	pub fn new(kernel_path: PathBuf, params: Params) -> HypervisorResult<UhyveVm<VCpuType>> {
 		let memory_size = params.memory_size.get();
 
+		// TODO: Use random address
+		let guest_address = arch::RAM_START;
+
 		#[cfg(target_os = "linux")]
-		let mem = MmapMemory::new(0, memory_size, arch::RAM_START, params.thp, params.ksm);
+		let mem = MmapMemory::new(
+			0,
+			memory_size,
+			GuestPhysAddr::new(guest_address),
+			params.thp,
+			params.ksm,
+		);
+
+		// TODO: guest_address is never taken to account on non-Linux platforms.
 		#[cfg(not(target_os = "linux"))]
-		let mem = MmapMemory::new(0, memory_size, arch::RAM_START, false, false);
+		let mem = MmapMemory::new(
+			0,
+			memory_size,
+			GuestPhysAddr::new(guest_address),
```

Comment:
For `#[cfg(not(target_os = "linux"))]`, this should always be `arch::RAM_START` until the other platforms are tested - especially given how we actively "don't care" about the `guest_address` here:

https://github.com/hermit-os/uhyve/pull/725/files#diff-460483abdc699825a37cf4f01218d714a3433f62cd8a4feda9c31911ffdc2beaR157